### PR TITLE
Cilium network policies

### PR DESCRIPTION
**Pull request by iainsproat**
## Description & motivation

Allows Cilium Network Policies to be provided if configured by the operator.

Cilium Network Policies provide more features compared to Kubernetes Network Policies, including DNS lookup and only allowing egress to specific domain names.

**Breaking change**: modifies the networkPolicy section of the values.yaml.  As we merged the network policy section only a few days ago, I'm confident it's unlikely to break anyone.

https://github.com/specklesystems/speckle-server/issues/913

## Changes:

- new helm chart value `networkPlugin.type` which toggles Network Policy between Kubernetes and Cilium.
- Cilium Network Policies for all services.
- FileImport uses a Kubernetes fully qualified domain name to contact the speckle-server service.
- small refactorings of the helm chart to make it cleaner and DRY.

## Checklist:

- [x] My pull request follows the guidelines in the [Contributing guide](https://github.com/specklesystems/speckle-server/blob/main/CONTRIBUTING.md)?
- [x] My pull request does not duplicate any other open [Pull Requests](../../pulls) for the same update/change?
- [x] My commits are related to the pull request and do not amend unrelated code or documentation.
- [x] My code follows a similar style to existing code.
- [x] I have added appropriate tests.
- [x] I have updated or added relevant documentation.

## References

https://docs.cilium.io/en/v1.9/policy/language/#egress

### CODE REVIEW

**Comment by gjedlicska** [comment]
did you add the new values key into the jsonschema?

**Comment by gjedlicska** [comment]
I think you forgot to add the new value to the jsonchema

**Comment by iainsproat** [comment]
I need to change the values a bit more (need to better support egress to dependencies which are hosted within the cluster) so this PR is still WIP.  Once it's done, I'll update the helm chart doc and json schema.

**Comment by iainsproat** [comment]
@gjedlicska - I updated the schema.json for the changes to date.  Will continue working on it today.

File: utils/test-deployment/run_tests.py [hunk 1]
```diff
@@ -51,9 +51,12 @@
 if not SERVER_VERSION:
     SERVER_VERSION = os.getenv('SERVER_VERSION')
 if SERVER_VERSION:
-    assert server_info.version == SERVER_VERSION, f"The deployed version {server_info.version} doesn't match the expected {SERVER_VERSION}"
-    print(f"Server version {SERVER_VERSION} is deployed and available")
+    if not SERVER_VERSION == 'latest':
```
Comment (iainsproat) [inline utils/test-deployment/run_tests.py:54]: Test blew up when I was using image `latest`, so this is an attempt at Python to fix that.

**Review by gjedlicska — APPROVED** [review]

**Pull request merged**